reconcile/gitlab_labeler.py

```diff
@@ -10,6 +11,60 @@
 QONTRACT_INTEGRATION = 'gitlab-labeler'
 
 
+def get_app_list() -> Dict:
+    """
+    Get a list of all services managed by app-interface with the parent app
+    and the onboarding status of the service.
+    """
+    apps = queries.get_apps()
+    app_info = {a['name']: {'onboardingStatus': a['onboardingStatus'],
+                'parentApp': a['parentApp']} for a in apps}
+    return app_info
+
+
+def get_parents_list() -> List[str]:
+    """
+    Get a set of all service names that are parents of another service
+    """
+    parent_set = set()
+    apps = queries.get_apps()
+    for a in apps:
+        if a['parentApp'] is not None:
+            parent_set.add(a['parentApp']['name'])
+
+    return list(parent_set)
+
+
+def guess_onboarding_status(changed_paths: Iterable[str],
+                            apps: Dict[str, Dict], parent_apps:
+                            Set[str]) -> Optional[str]:
+    """
+    Guess the service name of a given MR from the changed paths of the
+    MR. This will allow to add the onboarding status to the MR's as label
+    """
+    labels = set()
+    for path in changed_paths:
+        if 'data/services/' in path:
+            path_slices = path.split('/')
+            try:
+                if path_slices[2] in parent_apps:
+                    app = apps[path_slices[3]]
+                    if app is not None:
+                        labels.add(app['onboardingStatus'])
+                else:
+                    app = apps[path_slices[2]]
+                    if app is not None:
+                        labels.add(app['onboardingStatus'])
+                    else:
+                        return None
```

Comment:
This method assumes you will only be processing one app, why do you continue iterating over all `changed_paths` instead of `break`-ing here?

Reply:
As part of this effort we are trying to add the information only when we know that is 100% correct. 

We are iterating over all the different changed paths because we don't want to add partial information, If there are two different applications but we only add a tag for one, the information is incomplete.

Also, we don't want (or that was the idea)  to have two different onboarding status labels on a given MR that's why we are only adding it in case we have one application.

Reply:
@Piojo Added above the information about what we thought about the expected behavior on a multiple app MR

Reply:
Mmm... in that case I do believe this method is doing too many things. Identifying the app for each path and assigning the onboarding status are different things, and separating them will yield more obvious code.

Can you try that small refactor?